shopfloor/services/manual_product_transfer.py

```diff
@@ -0,0 +1,643 @@
+                return self._response_for_confirm_quantity(
+                    location, product, quantity, lot, message=self.msg_store.wrong_lot()
+                )
+            return
+        # Search by product
+        scanned_product = search.product_from_scan(barcode)
+        # Barcode is not a product
+        if not scanned_product:
+            return self._response_for_confirm_quantity(
+                location,
+                product,
+                quantity,
+                lot,
+                message=self.msg_store.no_product_for_barcode(barcode),
+            )
+        # Barcode is a product but doesn't match the processed one
+        if product != scanned_product:
+            return self._response_for_confirm_quantity(
+                location, product, quantity, lot, message=self.msg_store.wrong_product()
+            )
+
+    # FIXME copy pasted from location content transfer, put it elsewhere?
+    def _unreserve_other_lines(self, location, move_lines):
+        """Unreserve move lines in location in another picking type
+
+        Returns a tuple of (
+          move lines that stays in the location to process,
+          moves to reserve again,
+          response to return to client in case of error
+        )
+        """
+        lines_other_picking_types = move_lines.filtered(
+            lambda line: line.picking_id.picking_type_id not in self.picking_types
+        )
+        if not lines_other_picking_types:
+            return (move_lines, self.env["stock.move"].browse(), None)
+        unreserved_moves = move_lines.move_id
+        location_move_lines = self.env["stock.move.line"].search(
+            [
+                ("location_id", "=", location.id),
+                ("state", "in", ("assigned", "partially_available")),
+            ]
+        )
+        extra_move_lines = location_move_lines - move_lines
+        if extra_move_lines:
+            return (
+                self.env["stock.move.line"].browse(),
+                self.env["stock.move"].browse(),
+                self._response_for_start(
+                    message=self.msg_store.picking_already_started_in_location(
+                        extra_move_lines.picking_id
+                    )
+                ),
+            )
+        package_levels = move_lines.package_level_id
+        # if we leave the package level around, it will try to reserve
+        # the same package as before
+        package_levels.explode_package()
+        unreserved_moves._do_unreserve()
+        return (move_lines - lines_other_picking_types, unreserved_moves, None)
+
+    def scan_destination_location(self, move_line_ids, barcode):
+        """Scan the destination location and post the move.
+
+        Transitions:
+        * start: move has been posted successfully
+        * scan_destination_location: scanned location is wrong (error)
+        """
+        move_lines = self.env["stock.move.line"].browse(move_line_ids).exists()
```

Comment:
Should we not check that the location bar code scanned is the correct one ?

Reply:
Hey, but you are right! :D I totally missed this part...

Reply:
Fixed